cluster/src/main/java/com/hartwig/pipeline/PipelineMain.java

```diff
@@ -58,27 +60,30 @@ public static void main(String[] args) {
                     LOGGER.info("Single sample pipeline is complete with status [{}]. Stages run were [{}]", state.status(), state);
                 } else {
-                    PipelineState state =
-                            new SomaticPipeline(new AlignmentOutputStorage(storage, arguments, ResultsDirectory.defaultDirectory()),
-                                    new BamMetricsOutputStorage(storage, arguments, ResultsDirectory.defaultDirectory()),
-                                    SetMetadataApiProvider.from(arguments, storage).get(),
-                                    PipelineResultsProvider.from(storage, arguments, Versions.pipelineVersion()).get(),
-                                    new FullSomaticResults(storage, arguments),
-                                    CleanupProvider.from(credentials, arguments, storage).get(),
-                                    AmberProvider.from(arguments, credentials, storage).get(),
-                                    CobaltProvider.from(arguments, credentials, storage).get(),
-                                    SomaticCallerProvider.from(arguments, credentials, storage).get(),
-                                    StructuralCallerProvider.from(arguments, credentials, storage).get(),
-                                    PurpleProvider.from(arguments, credentials, storage).get(),
-                                    HealthCheckerProvider.from(arguments, credentials, storage).get(),
-                                    Executors.newCachedThreadPool()).run();
+                    state = new SomaticPipeline(new AlignmentOutputStorage(storage, arguments, ResultsDirectory.defaultDirectory()),
+                            new BamMetricsOutputStorage(storage, arguments, ResultsDirectory.defaultDirectory()),
+                            SetMetadataApiProvider.from(arguments, storage).get(),
+                            PipelineResultsProvider.from(storage, arguments, Versions.pipelineVersion()).get(),
+                            new FullSomaticResults(storage, arguments),
+                            CleanupProvider.from(credentials, arguments, storage).get(),
+                            AmberProvider.from(arguments, credentials, storage).get(),
+                            CobaltProvider.from(arguments, credentials, storage).get(),
+                            SomaticCallerProvider.from(arguments, credentials, storage).get(),
+                            StructuralCallerProvider.from(arguments, credentials, storage).get(),
+                            PurpleProvider.from(arguments, credentials, storage).get(),
+                            HealthCheckerProvider.from(arguments, credentials, storage).get(),
+                            Executors.newCachedThreadPool()).run();
                     LOGGER.info("Somatic pipeline is complete with status [{}]. Stages run were [{}]", state.status(), state);
                 }
+                if (state.status() == PipelineStatus.SUCCESS) {
```

Comment:
We could consider adding exit values to the `PipelineStatus` enum so we can just call `System.exit(state.status().code())` or something similar.

Reply:
See your point, would remove the need for the if, and look pretty tidy... but I think exits codes are more in the "domain" of PipelineMain than the status itself. Going to leave it as is for now.